includes/cli/class-amp-cli-validation-command.php

```diff
@@ -377,12 +400,237 @@ public function check_url( $args ) {
 			$url = $origin . '/' . ltrim( $url, '/' );
 		}
 
-		$result = AMP_Validation_Manager::validate_url( $url );
-		if ( $result instanceof WP_Error ) {
-			WP_CLI::error( $result );
+		$response = AMP_Validation_Manager::validate_url( $url );
+		if ( $response instanceof WP_Error ) {
+			WP_CLI::error( $response );
+		}
+
```

Comment:
Let's go ahead and store the results in the `amp_validated_url` post type by adding something like this:

```php
$validated_url_post = AMP_Validated_URL_Post_Type::store_validation_errors(
    $response['results'],
    $response['url'],
    wp_array_slice_assoc( $response, [ 'object' ] )
);
if ( $validated_url_post instanceof WP_Error ) {
    WP_CLI::error( $validated_url_post );
}
```

The URL to the results in the admin can then be another format that can be returned:

```php
if ( 'url' === $assoc_args['format'] ) {
    WP_CLI::line( get_permalink( $validated_url_post ) );
    exit( 0 );
}
```

And in general, at the very end of the method here prior to `exit` there can be printing to `STDERR` the admin URL, which would allow the JSON results to be returned in addition to the validated URL screen admin URL.